### PR TITLE
Shorten java code

### PR DESCRIPTION
**Pull request by m-kuhn**
This is a major change regarding

 - java bootstrap code
 - asset deployment

Almost all the initialization java code is gone. Java now does:

 - Creating the /QField/basemaps, /QField/proj, /QField/fonts directories
 - Obtaining some intent data for opening qgz files (to be tested)
 - Setting intent data for the app itself (version information, project file to open)

Assets are no longer zipped, they are delivered as files. They are copied from the original path to some user writable location on first run and when the version changes.

### CODE REVIEW

File: android/src/ch/opengis/qfield/QFieldActivity.java [hunk 1]
```diff
@@ -72,160 +72,33 @@
...
-    private void startQtActivity() {
         // forward to startQtActivity and finish QFieldActivity
         Intent intent = new Intent();
         intent.setClass(QFieldActivity.this, QtActivity.class);
-        intent.putExtra("DOTQGIS2_DIR", mDotQgis2Dir);
-        intent.putExtra("SHARE_DIR", mShareDir);
-        intent.putExtra("PACKAGE_PATH", getFilesDir().toString() + "/share");
-        intent.putExtra("QFIELD_DATA_DIR", mQFieldDir);
+        intent.putExtra("QFIELD_DATA_DIR", qFieldDir);
+        intent.putExtra("GIT_REV", mActivityInfo.metaData.getString("android.app.git_rev"));
```
Comment (m-kuhn) [inline android/src/ch/opengis/qfield/QFieldActivity.java:101]:
```suggestion
        intent.putExtra("GIT_REV", getPackageManager().getActivityInfo(getComponentName(), PackageManager.GET_META_DATA).metaData.getString("android.app.git_rev"));
```

File: android/src/ch/opengis/qfield/QFieldActivity.java [hunk 2]
```diff
@@ -72,160 +72,33 @@
...
-    private String mDotQgis2Dir;
-    private String mShareDir;
-    private String mQFieldDir;
-    private ProgressBar progressBar;
-    private TextView unpackingTitle;
-    private TextView unpackingMessage;
-    private static Application application;
-    private Dialog unpackingDialog;
-    private Dialog noExternalStorageDialog;
+    private ActivityInfo mActivityInfo = null; // activity info object, used to get git rev
```
Comment (m-kuhn) [inline android/src/ch/opengis/qfield/QFieldActivity.java:75]:
```suggestion
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c536f8cdaa8208275d31b05573de0083edbcb9da:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-1453c783c9e875f90ea7f2100690f158ca0e511f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-1453c783c9e875f90ea7f2100690f158ca0e511f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-1453c783c9e875f90ea7f2100690f158ca0e511f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-1453c783c9e875f90ea7f2100690f158ca0e511f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a4131b6b1db44a5ad2664c60cf8b8f1e137fc939:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-0fd2b620d266c5a8456e75cd4469bc74a09ce36a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-0fd2b620d266c5a8456e75cd4469bc74a09ce36a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-0fd2b620d266c5a8456e75cd4469bc74a09ce36a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-0fd2b620d266c5a8456e75cd4469bc74a09ce36a-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 57cef21ebe543a85733872b4f82f335ce1f6d60e:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-6b9cf965efb68daadc7bf7d77c8c3bb41f20204e-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-6b9cf965efb68daadc7bf7d77c8c3bb41f20204e-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-6b9cf965efb68daadc7bf7d77c8c3bb41f20204e-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-6b9cf965efb68daadc7bf7d77c8c3bb41f20204e-x86.apk)

**Comment by nirvn** [comment]
@m-kuhn , file association crashes now:

```
12-16 16:26:53.490   468   468 I QField  : Starting ...
12-16 16:26:53.490   468   468 I QField  : Opening project content://com.amaze.filemanager/storage_root/storage/emulated/0/QField/CambodiaSoldOut/CambodiaSoldOut_qfield.qgs
12-16 16:26:53.490   468   468 D AndroidRuntime: Shutting down VM
--------- beginning of crash
12-16 16:26:53.491   468   468 E AndroidRuntime: FATAL EXCEPTION: main
12-16 16:26:53.491   468   468 E AndroidRuntime: Process: ch.opengis.qfield_beta, PID: 468
12-16 16:26:53.491   468   468 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{ch.opengis.qfield_beta/ch.opengis.qfield_beta.QFieldActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context android.app.Application.getApplicationContext()' on a null object reference
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3654)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3806)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2267)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8167)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
12-16 16:26:53.491   468   468 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context android.app.Application.getApplicationContext()' on a null object reference
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at ch.opengis.qfield_beta.QFieldActivity.getContext(QFieldActivity.java:69)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at ch.opengis.qfield_beta.QFieldActivity.startQtActivity(QFieldActivity.java:106)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at ch.opengis.qfield_beta.QFieldActivity.onCreate(QFieldActivity.java:74)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7963)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:7952)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3629)
12-16 16:26:53.491   468   468 E AndroidRuntime: 	... 11 more
12-16 16:26:53.494  5022  7382 D Debug   : !@DumpState : SHIP
12-16 16:26:53.494  5022   949 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
12-16 16:26:53.494  5022  7382 D Debug   : !@DumpState : debug level:0x4f4c
12-16 16:26:53.494  5022  7382 D Debug   : !@Dumpstate : Finally, system will skip dumpstate
12-16 16:26:53.494  5022  7382 W ActivityManager: crash : ch.opengis.qfield_beta,0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger: Display 0 HWC layers:
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:     type    |    handle    | flag |  format   |   source crop (l,t,r,b)    |         frame       | name 
12-16 16:26:53.499  4553  4553 I SurfaceFlinger: ------------+--------------+------+-----------+----------------------------+---------------------+------
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f5e8a40c0 | 0002 | RGBA_8888 |   0.0   19.0 1080.0 2400.0 |    0    0 1080 2381 | com.amaze.filemanager/com.amaze.file[...].ui.activities.MainActivity$_32549#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:  SOLID_COLOR |              | 0004 |  Unknown  |   0.0    0.0    0.0    0.0 |    0    0 1080 2400 | Dim Layer for - Task=6177#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f5e8a2cc0 | 0002 | RGBA_8888 |   0.0    0.0 1080.0 2067.0 |    0  333 1080 2400 | Splash Screen ch.opengis.qfield_beta#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f60264300 | 0000 | RGBA_8888 |   0.0    0.0 1080.0   93.0 |    0    0 1080   93 | StatusBar$_5420#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f3fcc9500 | 0000 | RGBA_8888 |   0.0    0.0   67.0  429.0 | 1013  622 1080 1051 | com.samsung.android.app.cocktailbars[...]rservice.CocktailBarService$_12185#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f3fcc9c80 | 0000 | RGBA_8888 |   0.0 
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:    0.0 1080.0  126.0 |    0 2274 1080 2400 | NavigationBar0$_5420#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f5e8a1f00 | 0000 | RGBA_8888 |   0.0    0.0 1080.0  125.0 |    0    0 1080  125 | ScreenDecorOverlay$_5420#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger:      DEVICE | 0x7f60264d00 | 0000 | RGBA_8888 |   0.0    0.0 1080.0  100.0 |    0 2300 1080 2400 | ScreenDecorOverlayBottom$_5420#0
12-16 16:26:53.499  4553  4553 I SurfaceFlinger: 
12-16 16:26:53.499  5022  7382 W ActivityTaskManager:   Force finishing activity ch.opengis.qfield_beta/.QgsFileHandlerActivity
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c33804713b5579d118d3536c3b644a8f8cd0718d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-271388d03a7849ad09657649e03a2d71dad06db9-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-271388d03a7849ad09657649e03a2d71dad06db9-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-271388d03a7849ad09657649e03a2d71dad06db9-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-271388d03a7849ad09657649e03a2d71dad06db9-x86.apk)

**Comment by m-kuhn** [comment]
Works here now

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 4bbde024b5097c8d12e995829544be406599cdbc:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-82552edc62b1f97aeb9fda96b7570dc7e3de1699-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-82552edc62b1f97aeb9fda96b7570dc7e3de1699-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-82552edc62b1f97aeb9fda96b7570dc7e3de1699-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-82552edc62b1f97aeb9fda96b7570dc7e3de1699-x86.apk)

**Comment by m-kuhn** [comment]
whitescreen issue unresolved still.
But ready for merge.

@3nids any comments?

File: android/src/ch/opengis/qfield/QFieldActivity.java [hunk 3]
```diff
@@ -45,25 +45,19 @@
...
-import android.content.SharedPreferences;
 import android.content.pm.ActivityInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.PackageManager.NameNotFoundException;
 import android.net.Uri;
 import android.os.AsyncTask;
 import android.os.Bundle;
 import android.os.Environment;
+<<<<<<< HEAD
+=======
```
Comment (3nids) [inline android/src/ch/opengis/qfield/QFieldActivity.java:58]: Bad rebase

**Review by 3nids — APPROVED** [review]
Niiiiiiiiiice

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e273a4323fc5019f0e4c3eed220a119efd55ec06:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-5aa194ba1a9dc9755ffb93635ebd5967f9b5f976-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-5aa194ba1a9dc9755ffb93635ebd5967f9b5f976-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-5aa194ba1a9dc9755ffb93635ebd5967f9b5f976-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-5aa194ba1a9dc9755ffb93635ebd5967f9b5f976-x86.apk)

File: src/core/androidplatformutilities.cpp [hunk 4]
```diff
@@ -20,32 +20,38 @@
...
-QString AndroidPlatformUtilities::configDir() const
+void AndroidPlatformUtilities::initSystem()
 {
-  return getIntentExtra( "DOTQGIS2_DIR" );
-}
-
-QString AndroidPlatformUtilities::shareDir() const
-{
-  return getIntentExtra( "SHARE_DIR" );
+  QString appDataLocation = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation );
```
Comment (m-kuhn) [inline src/core/androidplatformutilities.cpp:41]:
```suggestion
  // Copy data away from the virtual path `assets:/` to a path accessible also for non-qt-based libs
  QString appDataLocation = QStandardPaths::writableLocation( QStandardPaths::AppDataLocation );
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1684b0f44894736a385fb307835ee2cf34c50969:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-7822fcb5bf39d4ebee32ee11edf821cd7d53e961-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-7822fcb5bf39d4ebee32ee11edf821cd7d53e961-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-7822fcb5bf39d4ebee32ee11edf821cd7d53e961-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1482-7822fcb5bf39d4ebee32ee11edf821cd7d53e961-x86.apk)

**Pull request merged**